Sources/Smithy/Context.swift

```diff
@@ -5,46 +5,63 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+import class Foundation.NSRecursiveLock
+
 public class Context {
-    public var attributes: Attributes
+    private var _attributes: Attributes
+    private let lock = NSRecursiveLock()
```

Comment:
`attributes` has been made private, and the lock enforces exclusive access to it.